### ISSUE TITLE
Bug with position in 360° equirectangular recording option

### ISSUE THREAD

**Issue by RYRY1002** [daipenger/minema#27]
# TLDR: All 6 camera positions in the 360° equirectangular recording options need to be moved back by `0.610` blocks.

A small change needs to be made to the 360° equirectangular recording option, the camera needs to be moved backwards by `0.610` blocks. This is because Minecraft's camera sphere _(The furthest away area that the game's camera can move to)_ is `0.610` blocks big. This is fine for normal recording, as it simulates your head being slightly ahead of your body's core, but for 360° recordings, it is a problem that needs to be fixed.

In case my words aren't clear enough, here is a demonstration that I took using version 3.6.1-ShaderSync.

### The coordinates for this first image are `X: 401.0, Y: 126.610, Z: -5.5`
![X: 401.0, Y: 126.610, Z: -5.5](https://cdn.discordapp.com/attachments/814066163075252230/864715164447277066/360Bug-126.61000.png)

As you can see in this image, the camera facing down is perfectly inside the block, despite the fact that it should be `0.610` above the block. This is how I determined that the camera needs to be moved back by `0.610` blocks.

### The coordinates for this second image are `X: 401.0, Y: 126.611, Z: -5.5`
![X: 401.0, Y: 126.611, Z: -5.5](https://cdn.discordapp.com/attachments/814066163075252230/864715173897699378/360Bug-126.61100.png)

Here I have moved the camera `0.001` blocks upwards, and now the camera is `0.001` blocks away from the block below the camera. Again, the Minema's camera position here is `0.611` blocks above the grass, but it appears `0.001` blocks away because of this bug.

You can also see in this image that the cobblestone block closest to the camera has been chopped in half. This is because this bug applies to all 6 rotations. This is also why none of the positions line up.

### The coordinates for this third image are `X: 401.5, Y: 126.8, Z: -5.5`
![X: 401.5, Y: 126.8, Z: -5.5](https://cdn.discordapp.com/attachments/814066163075252230/864715187423543306/360Bug-126.800.png)

This is a best-case scenario. The camera in Minema has been moved multiple blocks above the grass below it, so that hopefully the effect is less noticeable, however you can still definitely see the bug.

The cobblestone block from the second picture has also been fixed, however, this is because Minema's camera position has been set to be in the middle of the block, because of course, the edges of the frames are at 45° and so the edge of the block is all within one frame.

If you guys need any more info to fix this issue, please ask. I am happy to provide them.

**Comment by mchorse** [comment]
@NyaNLI you might want to take a look at this.

**Comment by ghost** [comment]
Sorry, no offense, but this issuse is caused by this code in your shaderpack
```glsl
position = gbufferModelView * position;
position.z += 0.1;
position = gbufferModelViewInverse * position;
```

And about the first image, this is because OpenGL discards samples with a depth less than a specific value, for Minecraft this value is 0.05 (after linearization). I can't fix it, sorry.

**Comment by mchorse** [comment]
Wow, that’s fascinating! Thanks for a quick reply, and I suppose it’s not Minema’s bug/issue. 

Although, @RYRY1002 thank you very much for such detailed overview in the original message. I suppose you can check whether same shader is having an issue with ReplayMod. If it doesn’t, I suppose it artificially raises camera position or something similar.

**Comment by RYRY1002** [comment]
Yeah, it's fine. This is a shaderpack I am making myself so I can change the code very easily.

**Comment by RYRY1002** [comment]
Yeah, thanks for the quick response, don't see that on most Minecraft mods.

Turns out I had implemented that exact code from before Minema had 360 because the mod didn't do that by itself.
Now, of course, it does. So what was happening was that Minema and my shader were both doing that code, overdoing it.

After removing the code, the issue was resolved.
Thanks for the help.

@NyaNLI deserves to be a collaborator I think.

**Comment by mchorse** [comment]
Awesome! I'm glad the issue was resolved! 🥳 

> @NyaNLI deserves to be a collaborator I think.

Indeed, however, I can't grant collaborator in this repository. Only @daipenger can. @NyaNLI if you want, I can give you collaborator role on mchorse/minema repo.

**Comment by ghost** [comment]
Thanks for your recognition, but it‘s not necessary I think. My name is already in the author list, so I think that's enough for me. :)
